### PR TITLE
Load versions from list.json

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -514,36 +511,52 @@ var run = function () {
...
+      loadVersion('builtin');
+      return;
+    }
+
+    //
+    // populate version dropdown with all available compiler versions
+    //
+
+    // NOTE: the list currently is oldest first
+    $.each(data.builds.reverse(), function (i, build) {
```
Comment (axic) [inline src/app.js]: Do we want to keep the builds in oldest first order?

**Comment by axic** [comment]
Do not merge yet, only after #216.

File: src/app.js (hunk 1 above)
Comment (chriseth) [inline src/app.js]: If we rename the files it does not matter, we can just sort them according to semver.

Comment (axic) [inline src/app.js]: So you want to sort them according to semver for the dropdown?

File: src/app.js [hunk 2]
```diff
@@ -526,8 +526,14 @@ var run = function () {
     // populate version dropdown with all available compiler versions
     //
 
-    // NOTE: the list currently is oldest first
-    $.each(data.builds.reverse(), function (i, build) {
+    // Sort builds according to semver
+    var builds = data.builds.sort(function (a, b) {
+      a = a.version + '-' + a.prerelease;
+      b = b.version + '-' + b.prerelease;
+      return semver.lt(a, b);
```
Comment (axic) [inline src/app.js]: Added sorting here, but it doesn't seem to work, the output is fairly random. Any ideas?

Comment (chriseth) [inline src/app.js]: Are you sure? My local tests look rather good.

File: src/app.js [hunk 3]
```diff
@@ -513,36 +510,58 @@ var run = function () {
...
-  if (window.soljsonSources !== undefined) {
-    // populate selector list with available versions
-    $.each(soljsonSources, function (i, file) {
-      if (file) {
-        var version = file.replace(/soljson-(.*).js/, '$1');
-        $('#versionSelector').append(new Option(version, file));
-      }
+  $.getJSON('https://ethereum.github.io/solc-bin/bin/list.json', function (data, status) {
+    //
+    // loading failed for some reason, fall back to local compiler
```
Comment (chriseth) [inline src/app.js]: Is this excessive spacing really required at that level?

File: src/app.js [hunk 4]
```diff
@@ -513,36 +510,58 @@ var run = function () {
...
+      return;
+    }
+
+    //
+    // populate version dropdown with all available compiler versions
+    //
+
+    // Sort builds according to semver
+    var builds = data.builds.sort(function (a, b) {
+      a = a.version + '-' + a.prerelease;
```
Comment (chriseth) [inline src/app.js]: In the future, `a.prerelease` might be empty. In that case, we also have to drop the `'-'`

File: src/app.js (hunk 2 above)
Comment (chriseth) [inline src/app.js]: What about `semver.compare(a, b)`? I think this function is required to return -1 / 0 / 1 and not true / false.

Comment (axic) [inline src/app.js]: That seems to do the trick.

File: src/app.js (hunk 3 above)
Comment (axic) [inline src/app.js]: Tried to separate the pieces with comments, reworked that a bit.

File: src/app.js (hunk 4 above)
Comment (axic) [inline src/app.js]: Have a method called `buildVersion()` for that now. With change of versioning, we need to update that method only.

File: src/app.js [hunk 5]
```diff
@@ -513,36 +510,58 @@ var run = function () {
...
+    // loading failed for some reason, fall back to local compiler
+    if (status !== 'success') {
+      $('#versionSelector').append(new Option('latest local version', 'builtin'));
+
+      loadVersion('builtin');
+      return;
+    }
+
+    function buildVersion (build) {
+      if (build.prerelease) {
```
Comment (chriseth) [inline src/app.js]: check for length?

File: src/app.js [hunk 6]
```diff
@@ -513,36 +510,58 @@ var run = function () {
...
+        return build.version;
       }
+    }
+
+    // Sort builds according to semver
+    var builds = data.builds.sort(function (a, b) {
+      a = buildVersion(a);
+      b = buildVersion(b);
+      return semver.compare(a, b);
+    }).reverse();
```
Comment (axic) [inline src/app.js]: I guess instead of `reverse()` I could use `1 - semver.compare(a, b)`, but this way it is more readable.

Comment (axic) [inline src/app.js]: Actually, swapping a/b should work

Comment (chriseth) [inline src/app.js]: Yes, but then please add a comment.

**Comment by axic** [comment]
Fixed #194.

**Pull request merged**